### PR TITLE
Implement RPC-Put-V1-24 (ResponseSchemaSpecifiedForSuccessStatusCode) and RPC-Async-V1-02 (ProvisioningStateSpecified)

### PR DESCRIPTION
**Pull request by AkhilaIlla**
Implement RPC-Put-V1-24 (ResponseSchemaSpecifiedForSuccessStatusCode)

### CODE REVIEW

File: packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+export const responseSchemaSpecifiedForSuccessStatusCode = (putOperation: any, _opts: any, ctx: any) => {
+  const errors = []
+  const path = ctx.path
+  const successCodes = ["200", "201"]
+
+  for (const code of successCodes) {
+    const response = putOperation.responses[code]
+    if (!response?.schema) {
```
Comment (jianyexi) [inline packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:8]: 201 is only required in long running put, so need to check if the 'response' is undefined?

Comment (rkmanda) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:8]: 201 is a valid status code for synchronous PUTs as well.

Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:8]: But we should not report error is it's undefined.

File: docs/rules.md [hunk 2]
```diff
@@ -56,13 +56,14 @@ Please refer to [api-version-parameter-required.md](./api-version-parameter-requ
...
-* -alpha
-* -beta
-* -rc (release candidate)
-* -privatepreview
+
+- -preview - Indicates the API version is in (public) preview
+- -alpha
+- -beta
+- -rc (release candidate)
+- -privatepreview
```
Comment (rkmanda) [inline docs/rules.md:67]: this seems unrelated

File: packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts (hunk 1 above)
Comment (rkmanda) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:8]: Thats true. This check should only throw if the schema is undefined "when 201 is present"

File: docs/rules.md (hunk 2 above)
Comment (jianyexi) [reply docs/rules.md:67]: yes, need to run `rush regen-ruleindex` to fix it

Comment (rkmanda) [reply docs/rules.md:67]: You might need to properly merge main to your branch

**Review by rkmanda — CHANGES_REQUESTED** [review]
:clock1:

File: docs/rules.md (hunk 2 above)
Comment (AkhilaIlla) [reply docs/rules.md:67]: I noticed that in Brett's PR too https://github.com/Azure/azure-openapi-validator/pull/424/files#diff-87f3af8240ee46430dbebe4e346fe9828cb1442f5175e470289fdf64183edc67
So, I thought that was something related to the project

File: packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts (hunk 1 above)
Comment (AkhilaIlla) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:8]: @rkmanda @jianyexi 
So, we dont have to check for 200 status code having a response schema specified, is that right?

Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:8]: You need check the response schema is specified only when the status code is present

File: docs/rules.md (hunk 2 above)
Comment (AkhilaIlla) [reply docs/rules.md:62]: The changes apart from mine are from "rush regen-ruleindex"

File: packages/rulesets/src/spectral/functions/lro-provisioning-state-specified.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      if (verb === "patch") {
+        codes = patchCodes
+      } else {
+        codes = putCodes
+      }
+
+      for (const code of codes) {
+        if (!getProperty(pathItem[verb].responses[code]?.schema, "provisioningState")) {
+          errors.push({
+            message: `${code} response in long running ${verb} operation is missing ProvisioningState property. A LRO PUT and PATCH operations response schema must have ProvisioningState specified. `,
```
Comment (jianyexi) [inline packages/rulesets/src/spectral/functions/lro-provisioning-state-specified.ts:26]:
```suggestion
            message: `${code} response in long running ${verb} operation is missing ProvisioningState property. A LRO PUT and PATCH operations response schema must have ProvisioningState specified.`,
```

File: packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
+export const responseSchemaSpecifiedForSuccessStatusCode = (putOperation: any, _opts: any, ctx: any) => {
+  const errors = []
+  const path = ctx.path
+  const successCodes = ["200", "201"]
+
+  for (const code of successCodes) {
+    if (putOperation.responses[code]) {
+      const response = putOperation.responses[code]
+      if (!response?.schema) {
```
Comment (jianyexi) [inline packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:9]: should report error only when 201 is present and schema is empty,

Comment (AkhilaIlla) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:9]: should we not check for 200?

Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:9]: Oh, I've updated the comment.

Comment (AkhilaIlla) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:9]: I am checking if the code 200/201 is present in line 7
and if code is present and schema isnt present (checked in line 9), we throw error right?

Comment (jianyexi) [reply packages/rulesets/src/spectral/functions/response-schema-specified-for-success-status-code.ts:9]: Okay, the 'response?.schema' misled me.
```suggestion
      if (!response.schema) {
```

**Review by jianyexi — APPROVED** [review]
